app/controllers/viewing_parties_controller.rb

```diff
@@ -0,0 +1,30 @@
+class ViewingPartiesController < ApplicationController
+  def new
+    @movie = Movie.find_by(api_id: params[:movie_id])
+    if @movie.nil?
+      @movie ||= Movie.create(title: params[:title],runtime: params[:runtime],api_id: params[:movie_id])
+    end
+  end
+
+  def create
+    @movie = Movie.find_by(api_id: params[:movie_id])
+    party = Party.create(user_id: current_user.id,
+                         movie_id: @movie.id,
+                         date: params[:date],
+                         time: params[:time],
+                         duration: params[:duration_of_party])
+
+    if party.save
+      Guest.create(party_id: party[:id], user_id: current_user.id, attending: true)
```

Comment:
I will play around with refactoring, because I think a party should have many guests and we can use that relationship.